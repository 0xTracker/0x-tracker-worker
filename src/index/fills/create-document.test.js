const timekeeper = require('timekeeper');

const createDocument = require('./create-document');
const V2_FILL = require('../../fixtures/fills/v2');
const V3_FILL = require('../../fixtures/fills/v3');

beforeAll(() => {
  timekeeper.freeze('2019-10-21T03:00:00.000Z');
});

afterAll(() => {
  timekeeper.reset();
});

it.skip('should create Elasticsearch document for fill', () => {
  const doc = createDocument(V2_FILL);
  expect(doc).toMatchSnapshot();
});

it.skip('should exclude value property when fill is unmeasured', () => {
  const fill = {
    ...V2_FILL,
    conversions: {
      ...V2_FILL.conversions,
      USD: { ...V2_FILL.conversions.USD, amount: null },
    },
  };
  const doc = createDocument(fill);

  expect(doc.value).toBeUndefined();
});

it.skip('assets should include bridgeAddress property when asset is bridged', () => {
  const fill = {
    ...V2_FILL,
    assets: V2_FILL.assets.map(asset => ({
      ...asset,
      bridgeAddress: '0x58b7b96e170e46c07d02fac903cd1b3356b7549f',
    })),
  };
  const doc = createDocument(fill);

  expect(doc.assets).toMatchSnapshot();
});

it.skip('should include any apps associated with the fill', () => {
  const fill = {
    ...V2_FILL,
    apps: [
      {
        appId: '8fc6beb5-3019-45f7-a55a-9a4c6b4b6513',
        type: 0,
      },
      {
        appId: '5067df8b-f9cd-4a34-aee1-38d607100145',
        type: 1,
      },
    ],
  };
  const doc = createDocument(fill);

  expect(doc.apps).toEqual([
    {
      id: '8fc6beb5-3019-45f7-a55a-9a4c6b4b6513',
      type: 0,
    },
    {
      id: '5067df8b-f9cd-4a34-aee1-38d607100145',
      type: 1,
    },
  ]);
});

it.skip('should halve trade volume for orderMatcher fill and set tradeCountContribution to 0.5', () => {
  const fill = {
    ...V2_FILL,
    relayerId: 2,
  };
  const doc = createDocument(fill);

  expect(doc.tradeVolume).toBe(329.44788459646225);
  expect(doc.tradeCountContribution).toBe(0.5);
});

it.skip('should index protocol fee for V3 fills', () => {
  const doc = createDocument(V3_FILL);

  expect(doc.protocolFeeETH).toBe(1500000000150000);
  expect(doc.protocolFeeUSD).toBe(0.338445000033844);
});
