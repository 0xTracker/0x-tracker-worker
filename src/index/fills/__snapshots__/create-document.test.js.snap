// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assets should include bridgeAddress property when asset is bridged 1`] = `
Array [
  Object {
    "bridgeAddress": "0x58b7b96e170e46c07d02fac903cd1b3356b7549f",
    "tokenAddress": "0x6b175474e89094c44da98b954eedeac495271d0f",
  },
  Object {
    "bridgeAddress": "0x58b7b96e170e46c07d02fac903cd1b3356b7549f",
    "tokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
  },
]
`;

exports[`should create Elasticsearch document for fill 1`] = `
Object {
  "assets": Array [
    Object {
      "bridgeAddress": undefined,
      "tokenAddress": "0x6b175474e89094c44da98b954eedeac495271d0f",
    },
    Object {
      "bridgeAddress": undefined,
      "tokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
    },
  ],
  "date": 2019-12-23T18:15:16.000Z,
  "feeRecipient": "0xa258b39954cef5cb142fd567a46cddb31a670124",
  "fees": Array [
    Object {
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
    },
    Object {
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
    },
  ],
  "maker": "0x6924a03bb710eaf199ab6ac9f2bb148215ae9b5d",
  "orderHash": "0xf186241e774ed0ecb6b6c101482f9a52d77caaf2db3165c9052f814e4c844f5e",
  "protocolFeeETH": undefined,
  "protocolFeeUSD": undefined,
  "protocolVersion": 2,
  "relayerId": 7,
  "senderAddress": "0x8018280076d7fa2caa1147e441352e8a89e1ddbe",
  "status": 1,
  "taker": "0x8018280076d7fa2caa1147e441352e8a89e1ddbe",
  "tradeCountContribution": 1,
  "tradeVolume": 658.8957691929245,
  "traders": Array [
    "0x6924a03bb710eaf199ab6ac9f2bb148215ae9b5d",
    "0x8018280076d7fa2caa1147e441352e8a89e1ddbe",
  ],
  "transactionHash": "0x36a35ae64def8beab677435be4762bf63977000172d47096230a94922383c856",
  "updatedAt": "2019-10-21T03:00:00.000Z",
  "value": 658.8957691929245,
}
`;
